tests/Unit/AddCustomProviderTest.php

```diff
@@ -25,9 +25,7 @@ public function tearDown()
     public function testIfApiProviderOnAuthWasSetCorrectly()
     {
         $request = Mockery::mock(Request::class);
-        $request->shouldReceive('all')->andReturn([
-            'provider' => 'companies',
-        ]);
+        $request->shouldReceive('get')->andReturn('companies')->with('provider');
 
         $middleware = new AddCustomProvider();
         $middleware->handle($request, function () {
```

Comment:
To improve the test coverage you can add tests simulating pass a `null` provider and pass a provider not existent on `config('auth.guards')`. E.g.:

```php
    public function testPassNotExistentProvider()
    {
        $this->expectException(OAuthServerException::class);

        $request = Mockery::mock(Request::class);
        $request->shouldReceive('get')->andReturn('not_found')->with('provider');

        $middleware = new AddCustomProvider();
        $middleware->handle($request, function () {
            return 'response';
        });
    }

    public function testDoNotPassProviderToRequest()
    {
        $this->expectException(OAuthServerException::class);

        $request = Mockery::mock(Request::class);
        $request->shouldReceive('get')->andReturn(null)->with('provider');

        $middleware = new AddCustomProvider();
        $middleware->handle($request, function () {
            return 'response';
        });
    }
```

Reply:
Good. Thanks.